src/EditorFeatures/Core.Wpf/StringIndentation/StringIndentationTaggerProvider.cs

```diff
@@ -25,120 +23,101 @@
 using Microsoft.VisualStudio.Text.Classification;
 using Microsoft.VisualStudio.Text.Editor;
 using Microsoft.VisualStudio.Text.Tagging;
-using Roslyn.Utilities;
 
-namespace Microsoft.CodeAnalysis.Editor.Implementation.StringIndentation
+namespace Microsoft.CodeAnalysis.Editor.Implementation.StringIndentation;
+
+/// <summary>
+/// This factory is called to create taggers that provide information about how strings are indented.
+/// </summary>
+[Export(typeof(IViewTaggerProvider))]
+[TagType(typeof(StringIndentationTag))]
+[VisualStudio.Utilities.ContentType(ContentTypeNames.CSharpContentType)]
+[VisualStudio.Utilities.ContentType(ContentTypeNames.VisualBasicContentType)]
+internal sealed partial class StringIndentationTaggerProvider : AsynchronousViewportTaggerProvider<StringIndentationTag>
```

Comment:
existing view taggers which used to do the math themselves to widen their viewport to determine what to tag can just subclass this new tagger.  

This is similar to how diagnostic tagging works where the core base tagger is responsible for splitting the work into syntax/semantic/analyzer-syntax/analzyer-semantic, and the subclasses care about their domain (squiggles, inline-errors, etc).  

Here though the core tagger is responsible for tagging the visible view and the non-visible parts, and the subclasses just deal with actually producing the tags for those regions when asked.